### PR TITLE
Get role assignments for a specific resource & role name

### LINKED ISSUE #375: List assigned roles by resource & role name
needed for #364

### PR DESCRIPTION
**Pull request by sbose78**
This API returns all assignees to a specific role, for a specific resource.

Request:
```
GET /api/resources/:resourceID/roles/:roleName
Authorization: Bearer <User OpenShift.io token>
```

A 200OK Response would look like :
```
{
  data: [
    {
      assignee_id: "cb567f71-b09c-4784-a692-9270a6679af8",
      assignee_type: "user",
      inherited: false,
      role_name: "collab"
    },
    {
      assignee_id: "bb577f71-b09c-4784-a692-9270a6679af8",
      assignee_type: "user",
      inherited: false,
      role_name: "collab"
    },
    {
      assignee_id: "af567f71-b09c-4784-a692-9270a6679af8",
      assignee_type: "user",
      inherited: false,
      role_name: "collab"
    }
  ]
}
```
fixes #375

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-374-1`

**Comment by alexeykazakov** [comment]
Please fix the link to the issue and update the description of the issue accordingly.

**Comment by alexeykazakov** [comment]
Please add more details to the description.
This endpoint returns the list of all identities with roles which have any role assigned to this resource. Right? 
Make it clear that the JSON in this description is the result. Not request payload. Add Response status. Etc.

File: authorization/role/model/role_management_model_service.go [hunk 1]
```diff
@@ -34,6 +35,121 @@ type GormRoleManagementModelService struct {
...
+
+	columns, err := rows.Columns()
+	columnValues := make([]interface{}, len(columns))
+
+	var ignore interface{}
+	for index := range columnValues {
+		columnValues[index] = &ignore
+	}
+
+	if err != nil {
```
Comment (alexeykazakov) [inline authorization/role/model/role_management_model_service.go]: Move the error check up. Just after `columns, err := rows.Columns()`

File: authorization/role/model/role_management_model_service.go [hunk 2]
```diff
@@ -34,6 +35,121 @@ type GormRoleManagementModelService struct {
...
+		return identityRoles, err
+	}
+	defer rows.Close()
+
+	columns, err := rows.Columns()
+	columnValues := make([]interface{}, len(columns))
+
+	var ignore interface{}
+	for index := range columnValues {
+		columnValues[index] = &ignore
```
Comment (alexeykazakov) [inline authorization/role/model/role_management_model_service.go]: Why do you need this?

File: authorization/role/model/role_management_model_service.go (hunk 1 above)
Comment (sbryzak) [reply authorization/role/model/role_management_model_service.go]: @alexeykazakov possibly yes, but we can decide later whether we want to return mapped roles also.

Comment (sbose78) [reply authorization/role/model/role_management_model_service.go]: yeah, guessing just the query would change then.

File: authorization/role/model/role_management_model_service.go (hunk 2 above)
Comment (sbose78) [reply authorization/role/model/role_management_model_service.go]: was using it to explicitly initialize ( have this in a number of places in our code ) - amazingly it works now without this. ( the make function does the initialization it seems ) 
65b4b61

File: authorization/role/model/role_management_model_service.go (hunk 1 above)
Comment (sbose78) [reply authorization/role/model/role_management_model_service.go]: 65b4b61

**Comment by sbose78** [comment]
65b4b61
Explicitly added `Unauthorized` as a possible response status

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-374-2`

File: controller/resource_roles_blackbox_test.go [hunk 3]
```diff
@@ -138,6 +143,74 @@ func (rest *TestResourceRolesRest) TestListAssignedRolesFromInheritedOK() {
 	require.True(rest.T(), rest.checkExists(*identityRoleRef2, returnedIdentityRoles, true))
 }
 
+func (rest *TestResourceRolesRest) TestListAssignedRolesByRoleNameFromInheritedOK() {
```
Comment (sbryzak) [inline controller/resource_roles_blackbox_test.go]: Nice tests!

File: design/role.go [hunk 4]
```diff
@@ -17,6 +17,18 @@ var _ = a.Resource("resource_roles", func() {
 		a.Response(d.OK, identityRolesMedia)
 		a.Response(d.InternalServerError, JSONAPIErrors)
 		a.Response(d.NotFound, JSONAPIErrors)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+	})
+	a.Action("listAssignedByRoleName", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET("/:resourceID/roles/:roleName/assigned"),
```
[the remark this answers is not in the document]
Comment (sbose78) [reply design/role.go]: Makes absolute sense, will refactor :)

[the remark this answers is not in the document]
Comment (sbose78) [reply design/role.go]: So this is what I have now

```
action=ListAssigned ctrl=ResourceRoles route="GET /api/resources/:resourceID/roles" security=jwt
action=ListAssignedByRoleName ctrl=ResourceRoles route="GET /api/resources/:resourceID/roles/:roleName" security=jwt
```

0730ad8

[the remark this answers is not in the document]
Comment (sbryzak) [reply design/role.go]: Looks good!

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-374-3`

**Pull request merged**